pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceTransaction.java

```diff
@@ -0,0 +1,613 @@
+        public String adminURL;
+
+
+        @Parameter(names = {"-u", "--service-url"}, description = "Pulsar Service URL")
+        public String serviceURL;
+
+
+        @Parameter(names = {"-np",
+                "--partitions"}, description = "Create partitioned topics with the given number of partitions, set 0 "
+                + "to not try to create the topic")
+        public Integer partitions = null;
+
+        @Parameter(names = {"-c",
+                "--max-connections"}, description = "Max number of TCP connections to a single broker")
+        public int maxConnections = 100;
+
+        @Parameter(names = {"-time",
+                "--test-duration"}, description = "Test duration in secs. If 0, it will keep publishing")
+        public long testTime = 0;
+
+
+        @Parameter(names = {"-ioThreads", "--num-io-threads"}, description = "Set the number of threads to be " +
+                "used for handling connections to brokers, default is 1 thread")
+        public int ioThreads = 1;
+
+        @Parameter(names = {"-ss",
+                "--subscriptions"}, description = "A list of subscriptions to consume on (e.g. sub1,sub2)")
+        public List<String> subscriptions = Collections.singletonList("sub");
+
+        @Parameter(names = {"-ns", "--num-subscriptions"}, description = "Number of subscriptions (per topic)")
+        public int numSubscriptions = 1;
+
+
+        @Parameter(names = {"-sp", "--subscription-position"}, description = "Subscription position")
+        private SubscriptionInitialPosition subscriptionInitialPosition = SubscriptionInitialPosition.Latest;
+
+
+        @Parameter(names = {"-st", "--subscription-type"}, description = "Subscription type")
+        public SubscriptionType subscriptionType = SubscriptionType.Exclusive;
+
+        @Parameter(names = {"-q", "--receiver-queue-size"}, description = "Size of the receiver queue")
+        public int receiverQueueSize = 1000;
+
+
+        @Parameter(names = {"-timeout", "--txn-timeout"}, description = "transaction timeout")
+        public long transactionTimeout = 5;
+
+        @Parameter(names = {"-nmp", "--numMessage-perTransaction-produce"},
+                description = "the number of messages produced by  a transaction")
+        public int numMessagesProducedPerTransaction = 1;
+        @Parameter(names = {"-nmc", "--numMessage-perTransaction-consume"},
+                description = "the number of messages consumed by  a transaction")
+        public int numMessagesReceivedPerTransaction = 1;
+
+        @Parameter(names = {"-ntnx",
+                "--number-txn"}, description = "the number of transaction, if o, it will keep opening")
+        public long numTransactions = 0;
+
+        @Parameter(names = {"-txn", "--txn-enable"}, description = " whether transactions need to  be opened ")
+        public boolean isEnableTransaction = false;
+
+        @Parameter(names = {"-end"}, description = " how to end a transaction, commit or abort")
```

Comment:
Whether to commit or abort the transaction. (Only "--txn-enable true" can it take effect) is better